node/sc/bridge_accounts.go

```diff
@@ -146,12 +170,13 @@ func (acc *accountInfo) GetTransactOpts() *bind.TransactOpts {
 	if acc.isNonceSynced {
 		nonce = new(big.Int).SetUint64(acc.nonce)
 	}
-	return bind.MakeTransactOptsWithKeystore(acc.wallet, acc.address, nonce, acc.chainID, DefaultBridgeTxGasLimit, acc.gasPrice)
+
+	return bind.MakeTransactOptsWithKeystore(acc.keystore, acc.address, nonce, acc.chainID, DefaultBridgeTxGasLimit, acc.gasPrice)
 }
 
 // SignTx signs a transaction with the accountInfo.
 func (acc *accountInfo) SignTx(tx *types.Transaction) (*types.Transaction, error) {
-	return acc.wallet.SignTx(accounts.Account{Address: acc.address}, tx, acc.chainID)
+	return acc.keystore.SignTx(accounts.Account{Address: acc.address}, tx, acc.chainID)
```

Comment:
Don't we need a lock to protect the keystore access?

Reply:
Yes. To manage the account nonce, before calling this SignTx, the account lock is locked. After this function, it will be unlocked.